### ISSUE TITLE
Allow groups to specify which BioProject we upload their sequences into

### ISSUE THREAD

**Issue by anna-parker** [loculus-project/loculus#3466]
By default we should still create a bioproject for each (group, organism), but we should also let users give a bioproject accession that we can upload sequences to.

At the moment we can hack this with one transaction in the ena DB. In future we should add this as a field when users upload sequences for a specific organism and they should be able to see it on their group page. We will also need to be able to update this in the deposition pod.

**Comment by anna-parker** [comment]
Steps: 

1. Allow input of bioproject and biosample (remove `noInput` in values.yaml), potentially add description to tell users when to add this
2. Modify the DB table PROJECT_table to not use (group_id, organism) as unique ID -> small DB migration
3. Modify the SUBMISSION_table code to store PROJECT_table's new id for new created bioproject
4. Modify code: if bioproject exists create a new entry in PROJECT_table with the details - do not create a new bioproject, else create new bioproject for (group_id, organism) as before
5. Modify code: if biosample exists create a new entry in SAMPLE_table with details - do not create a new biosample, else create a new biosample as before

**Comment by emmahodcroft** [comment]
Ok, so the idea is that users can use these fields, but that we don't really surface/publicise it, and leave it for advanced users/those who ask. 
We should maybe clarify in the docs that if you provide a biosample we do not send on any metadata so it relies on them having the metadata they want in the biosample!

**Comment by emmahodcroft** [comment]
We should also check if the same thing will happen as did with the sraAccession - we had a check in place that we don't upload to ENA if this field is present to prevent double-submission - that check may also be present on these fields, so we will also have to allow upload even if present?

**Comment by theosanderson** [comment]
This concept of accepting metadata that we don't send feels like a big change for me: it breaks the concept of INSDC synchronisation it seems like? IMO that's a real hack that we should only deploy when strictly necessary because of a request from a user and not something that should be a typical practice or advertised. IMO it's very rare that the order of upload is raw reads then consensus (though clearly it does happen)

**Comment by anna-parker** [comment]
> We should also check if the same thing will happen as did with the sraAccession - we had a check in place that we don't upload to ENA if this field is present to prevent double-submission - that check may also be present on these fields, so we will also have to allow upload even if present?

I modified this in https://github.com/loculus-project/loculus/pull/3647 - we should be notified of the sequences with external metadata fields now - I will change the message once these changes have merged

**Comment by anna-parker** [comment]
> This concept of accepting metadata that we don't send feels like a big change for me

I don't think this is an issue - if a user adds these fields it implies the data is already on the INSDC. There might be some differences in metadata field mappings - but we already have that issue because we use different metadata fields than the INSDC.

**Comment by theosanderson** [comment]
Am I not understanding correctly that if users submit slightly different data to INSDC and then to Pathoplexus we will have an inconsistency that persists indefinitely? Until now we could guarantee that an open or ingested Pathoplexus sequence had the same data, wether accessed through INSDC or PP.

**Comment by anna-parker** [comment]
@theosanderson maybe we should have a quick call about this. Currently no - we try to sync the data on Pathoplexus with that on INSDC but it is not the same because we have different metadata fields as the INSDC and also not all data is available via the datasets CLI. When we upload to ENA we try to map to their fields but not all data is uploaded as not all data can be mapped.

The main reason we don't want users to submit to ENA independently is that  the data will then exist twice on INSDc and also pn Pathoplexus as we will reingest the data - however if we use the bioproject and biosample they already created we will not reingest this data and the data will also not be duplicated on the INSDC.

**Comment by theosanderson** [comment]
Yep but we do have a metadata synchronization procedure, such that our standard fields, dates, locations, etc. will be synced up. We have a procedure for stuff external people upload to INSDC to come into PPX, and for updates at INSDC to trigger updates on PPX, and we have a system for stuff submitted to Pathoplexus to go to INSDC and for updates to be propagated to INSDC. Here we could end up with explicitly contradictory metadata in the two places.

**Comment by anna-parker** [comment]
>  Here we could end up with explicitly contradictory metadata in the two places.

I don't think this is too large an issue but I agree there is a small chance that there could be worse metadata contradictions than there currently are. (I still think having 2 biosamples for the same sequence which are unlinked would be worse).

To avoid this edge case maybe we tell users that we would prefer that they submit consensus sequences to us first and then the raw reads. We will then minimize the likelihood of this edge case but still be able to handle it if it does come up.

**Comment by theosanderson** [comment]
Yeah, sgtm, sort of the lines I was thinking above:

> IMO that's a real hack that we should only deploy when strictly necessary because of a request from a user and not something that should be a typical practice or advertised.
